src/controller/AdminController.php

```diff
@@ -0,0 +1,267 @@
+<?php
+
+namespace Projet5\controller;
+
+use Exception;
+use Projet5\model\CommentModel;
+use Projet5\model\PostModel;
+use Projet5\model\UserModel;
+
+/**
+ * management of validations by admin 
+ */
+class AdminController extends SessionController
+{
+	/**
+	 * retrieve information awaiting validation by admin 
+	 *
+	 * @param UserModel $userModel
+	 * @param PostModel $postModel
+	 * @param CommentModel $commentModel
+	 *
+	 * @throws LoaderError
+	 * @throws RuntimeError
+	 * @throws SyntaxError
+	 * @throws Exception
+	 */
+	public function displayAllElements(UserModel $userModel, PostModel $postModel, CommentModel $commentModel)
+	{
+		// If I do not follow admin, return to the article page 
+		if ($_SESSION['rankConnectedUser'] !== 'admin') {
+			$_SESSION['error'] = 'Cette page est réservé à l\'administrateur';
+			header('location:Articles-Page1');
+			exit;
+		}
+
+		// controle if a POST variable exist and execute
+		$this->controleForms($userModel, $postModel, $commentModel);
+
+		// load pending users
+		$pendingUsers = $userModel->loadPendingUsers();
+		// load invalide posts
+		$invalidePosts = $postModel->loadAllPost($valide = 'waiting');
+		// load invalide comments
+		$invalideComments = $commentModel->loadInvalidComments();
+		// load the refused comments
+		$refuseComments = $commentModel->loadRefuseComments();
+
+		// display 
+		echo $this->twig->render('admin.twig', [
+			'SESSION' => $_SESSION,
+			'pendingUsers' => $pendingUsers,
+			'invalidePosts' => $invalidePosts,
+			'invalideComments' => $invalideComments,
+			'refuseComments' => $refuseComments
+		]);
+	}
+
+	/**
+	 * retrieve posts pending administrator validation 
+	 *
+	 * @param UserModel $userModel
+	 * @param PostModel $postModel
+	 * @param CommentModel $commentModel
+	 *
+	 * @throws LoaderError
+	 * @throws RuntimeError
+	 * @throws SyntaxError
+	 * @throws Exception
+	 */
+	public function displayWaitingPosts(UserModel $userModel, PostModel $postModel, CommentModel $commentModel)
+	{
+		// If I do not follow admin, return to the article page 
+		if ($_SESSION['rankConnectedUser'] !== 'admin') {
```

Comment:
```suggestion
		if ($_SESSION['rankConnectedUser'] !== UserModel::USER_RIGHT_ADMIN) {
```

et créer une constante dans ton UserModel

Reply:
constante ajoutée